src/main/java/org/truffleruby/core/encoding/EncodingConverterNodes.java

```diff
@@ -479,8 +478,8 @@ protected Object setReplacement(RubyEncodingConverter encodingConverter, Object
         }
 
         @TruffleBoundary
-        private int setReplacement(EConv ec, byte[] string, int len, byte[] encodingName) {
-            return ec.setReplacement(string, 0, len, encodingName);
+        private int setReplacement(EConv ec, InternalByteArray byteArray, byte[] encodingName) {
```

Comment:
This forces the allocation of the `InternalByteArray` so it's a bit suboptimal.
So it seems better to TruffleBoundary only ec.setReplacement and pass `byte[], int, int` separately.

Reply:
I've done that.